gazelle_cabal/dependency_resolution.go

```diff
@@ -224,8 +236,10 @@ func getPackageLabel(
 			// Cabal produces an error for circular dependency
 			log.Fatalf("Dependency cycle detected in the following component: %s", from)
 		}
-		// if it's indeed internal library than take it
-		if r.Label.Repo == from.Repo && r.Label.Pkg == from.Pkg {
+
+		// indeed internal library can be refereced form the package of its definition
+		// or it can be referenced using experimental colon syntax (a.k.a public internal libraries)
+		if (r.Label.Repo == from.Repo && r.Label.Pkg == from.Pkg) || (len(parentPackage) > 0) {
```

Comment:
Note the typos in the comment.

Still, is this the right behavior? If a dependency string refers to a sublibrary of another package, this code will chose instead an internal library.

Reply:
I probably don't get your question :/

The problem I see now is that I must check value of `parentPackage`. That's because there can be a situation where e.g. two or more packages will define a sublibrary with the same name, e.g. `package-a:sublib`, `package-b:sublib`, `package-c:sublib`. My code here if unfortunately ignoring left hand side of the colon syntax so `package-a:sublib` can be resolved with this code to `//package-b:sublib`

The first part of this `if` statement is responsible for `shadowing` feature - if we detect a dependency which resolves to some 'local definition' then, we get this local definition from `internal_library` pool rather then from `@stackage` repository.

Second part takes a dependency from `internal_library` pool only if it was found there and if it was referenced with colon syntax - but for sure I need to fix ambiguity case there.

Reply:
I think you get it.

The case I was referring to is, given that you are resolving dependencies of `package-a`, and that one such dependency is `package-b:sublib`, if `package-a` contains an internal library called `sublib`, this code will chose the internal library instead of the intended sub-library.